lib/player.rb

```diff
@@ -0,0 +1,36 @@
+class Player
+  attr_reader :score, :inputs, :mark
+
+  def initialize (mark)
+    @mark = mark
+    @score = 0
+    @inputs = []
+  end
+
+  def save (input)
+    @inputs << input
+  end
+
+  def valid? (input)
+    if 1 <= input && input <= 9
+      @inputs.each { |x| return false if input == x }
+    else
+      return false
+    end
+    true
+  end
+
+  def victory?
+    v_conditions = [[1,2,3], [4,5,6], [7,8,9], [1,4,7], [2,5,8], [3,6,9], [1,5,9], [3,5,7]]
+    v_conditions.each { |x| return true if @inputs.any?(x[0]) && @inputs.any?(x[1]) && @inputs.any?(x[2]) }
+    false
+  end
+
+  def reset
```

Comment:
If this method resets the player's moves whenever the game is reset then I don't think it should be the player's responsibility to reset his/her own moves/inputs array. When the game is reset (so it's really important to have a Game class) then all the objects in the game (player1, player2 and board) should all be re-initialized.